app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ComponentType.scala

```diff
@@ -97,14 +97,48 @@ case class Address(international: Boolean) extends ComponentType {
 
 }
 
-case object Address {
+object Address {
   val mandatoryFields: FormComponentId => List[FormComponentId] = id => List("street1").map(id.withSuffix)
   val optionalFields: FormComponentId => List[FormComponentId] = id =>
     List("street2", "street3", "street4", "uk", "postcode", "country").map(id.withSuffix)
   val fields: FormComponentId => NonEmptyList[FormComponentId] = id =>
     NonEmptyList.fromListUnsafe(mandatoryFields(id) ++ optionalFields(id))
 }
 
+case class OverseasAddress(
+  mandatoryFields: List[OverseasAddress.Mandatory],
+  value: Option[OverseasAddress.Value]
+) extends ComponentType
+
+object OverseasAddress {
+
+  sealed trait Mandatory
+
+  object Mandatory {
+    case object Line2 extends Mandatory
```

Comment:
Usual Line1 is mandatory? Why do we have Line2?

Reply:
This represents fields which are mandatory by choice.

From the ticket:
> Line 1, City and Country are mandatory

Reply:
Sorry, just read. It says 

`Line 1, City and Country are mandatory` 

which means we dont need to have the configurable flag for them? Why is City in there if it's mandatory always? Also, are we missing Line3 here?

Reply:
Ok, seems like I've misunderstood the ticket and implemented different semantics.